### ISSUE TITLE
Curious about the code of clustering visualization of different blocks

### ISSUE THREAD

**Issue by hollow-503** [NVlabs/FAN#3]
Thanks for your interesting work on Vit, which provides insight into Vit. After reading your paper, I am curious about the code of clustering visualization of different blocks. Could you release the code about that?
![image](https://user-images.githubusercontent.com/57223231/168772266-b2b16b53-12a8-4320-ae18-abdfe9435cfc.png)

**Comment by hollow-503** [comment]
Is that means I need to use spectral clustering to classify the vectors in each token into two classes(background and foreground), and then feed it into visualization code from the DINO repo?

**Comment by zhoudaquan** [comment]
> Is that means I need to use spectral clustering to classify the vectors in each token into two classes(background and foreground), and then feed it into visualization code from the DINO repo?

Yeah, we do not differentiate between the foregraound and background objects. Directly feed the generated mask into visualization code will do the job.

**Comment by hollow-503** [comment]
Oh, i see.I should set n_clusters=2 to generate the mask. Am i right?

**Comment by hollow-503** [comment]
Truly appreciate your help! I will try it first.
